plugin-server/tests/job-queue-manager.test.ts

```diff
@@ -0,0 +1,70 @@
+import { RetryError } from '@posthog/plugin-scaffold'
+
+import { runRetriableFunction } from '../src/utils/retries'
+import { UUID } from '../src/utils/utils'
+import { PromiseManager } from '../src/worker/vm/promise-manager'
+import { JobQueueManager } from './../src/main/job-queues/job-queue-manager'
+import { EnqueuedJob, Hub, JobName } from './../src/types'
+
+jest.mock('../src/utils/retries')
+
+const mockHub: Hub = {
+    instanceId: new UUID('F8B2F832-6639-4596-ABFC-F9664BC88E84'),
+    promiseManager: new PromiseManager({ MAX_PENDING_PROMISES_PER_WORKER: 1 } as any),
+    JOB_QUEUES: 'fs',
+} as Hub
+
+describe('JobQueueManager', () => {
+    let jobQueueManager: JobQueueManager
+    beforeEach(() => {
+        jest.clearAllMocks()
+        jobQueueManager = new JobQueueManager(mockHub)
+    })
+
+    describe('enqueue()', () => {
```

Comment:
Missing a test for error handling.

Reply:
There's tests for _enqueue error handling and error handling in the retries flow. This function just calls both of those, what do you mean?